### ISSUE TITLE
10$: Allow 'break await;' in non-async functions, and 'break yield' in non-generator functions

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#369]
False positive:
```js
function f() {
    await: for(;;) {
        break await;
    }
}
```

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.
